builtin/providers/aws/resource_aws_api_gateway_base_path_mapping_test.go

```diff
@@ -0,0 +1,137 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+	//"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSAPIGatewayBasePath_basic(t *testing.T) {
+	var conf apigateway.BasePathMapping
+	name := "test-api-gateway-domain-cert0.com"
```

Comment:
I think we're going to need to put some randomisation here as the domain seems to be globally unique in AWS. Kinda like S3 bucket names. The test has failed for me:

```
TF_ACC=1 go test ./builtin/providers/aws -v -run=AWSAPIGatewayBase -timeout 120m
=== RUN   TestAccAWSAPIGatewayBasePath_basic
--- FAIL: TestAccAWSAPIGatewayBasePath_basic (19.88s)
       	testing.go:265: Step 0 error: Error applying: 1 error(s) occurred:

       		* aws_api_gateway_domain_name.test: Error creating API Gateway Domain Name: BadRequestException: The domain name you provided is already associated with an existing CloudFront distribution.
       		Remove the domain name from the existing CloudFront distribution or use a different domain name.
       		If you own this domain name and are not using it on an existing CloudFront distribution, please contact support.
       			status code: 400, request id: 9******c-6852-11e6-9b0c-df6fa******2
```

I reckon it's because someone (you) created this domain in AWS and forgot to delete it.